crates/text-parser/src/grammar.lalrpop

```diff
@@ -258,8 +258,10 @@ pub(crate) OutgoingBindingExpression: A::OutgoingBindingExpression = {
 };
 
 OutgoingBindingExpressionAs: A::OutgoingBindingExpressionAs =
-    "(" "as" <ty:WebidlTypeRef> <idx:Unsigned> ")" =>
-        actions.outgoing_binding_expression_as(ty, idx);
+    "(" "as" <ty:WebidlTypeRef?> <idx:Unsigned> ")" => {
+        let mut actions_copy = actions.clone();
+        actions.outgoing_binding_expression_as(ty.unwrap_or_else(|| actions_copy.webidl_default_type_ref().into()), idx)
```

Comment:
I don't think we want to require that `Actions` be clone, since the default AST-creating actions have an internal arena and we don't want to have to wrap everything up in an `Rc` so that we avoid cloning the arena.